### PR TITLE
Refactor user utility functions to use data-access layer

### LINKED ISSUE #2147: Fix user utils file to use data access layer
### Issue Description
In the user utils file, we need to refactor some functions to use the data access layer to avoid circular dependency and other issue

### Reproducibility

 - [ ] This issue is reproducible
 - [ ] This issue is not reproducible

### Steps to Reproduce
1. 
2. 

### Severity/Priority
 - [ ] Critical
 - [ ] High
 - [ ] Medium
 - [ ] Low

### Checklist

- [ ] I have read and followed the project's code of conduct.
- [ ] I have searched for similar issues before creating this one.
- [ ] I have provided all the necessary information to understand and reproduce the issue.
- [ ] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by pankajjs**
Date: 8 Oct, 2024

Developer Name: Pankaj Sha

---

## Issue Ticket Number

- closes #2147 
## Description

This PR refactors user's utility functions to use data-access layer.
### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Test Coverage
<details>
<summary>Screenshot 1</summary>

Screenshot 2024-10-08 at 1 03 23 PM
</details>

### CODE REVIEW

File: utils/users.js
[the remark this answers is not in the document]
Comment (pankajjs) [reply utils/users.js:1]: @vinit717 It is being used in getUserName function. These tests are breaking while refactoring it.
Screenshot 2024-10-08 at 4 58 36 PM

[the remark this answers is not in the document]
Comment (vinit717) [reply utils/users.js:1]: Bu we need to refactor as well 
for this you'll also need to refactor the tests

**Comment by vinit717** [comment]
Can you add the screenshot of API response which uses the function you have changed

**Comment by pankajjs** [comment]
> Can you add the screenshot of API response which uses the function you have changed

I have made changes in retrieveUsers of dataAccessLayer. It has different responses based on the parameter.
For the null case, which I have handled the response look like this - 
`
{
userExists: false,
user: {
id: undefined
}
}
`

File: models/auctions.js [hunk 1]
```diff
@@ -27,7 +27,6 @@ const fetchAuctionById = async (auctionId) => {
       return { ...bidData, bidder: bidderUsername };
     });
     const biddersAndBids = await Promise.all(promises);
-
```
Comment (vinit717) [inline models/auctions.js:30]: Please remove this change also

Comment (pankajjs) [reply models/auctions.js]: @vinit717 Removed.

**Review by vinit717 — APPROVED** [review]

**Pull request merged**